apps/editing-toolkit/editing-toolkit-plugin/wpcom-coming-soon/class-wpcom-coming-soon.php

```diff
@@ -0,0 +1,188 @@
+	}
+
+	/**
+	 * Excludes sitemap items by post ID.
+	 *
+	 * @since 2018-05-04
+	 *
+	 * @param  bool               $exclude True if excluding.
+	 * @param  \WP_Post|\StdClass $post    Post object associated with the item.
+	 *
+	 * @return bool                        Filtered exclusion. True if excluding.
+	 *
+	 * @internal Site owners can add a custom field '_sitemap_exclude` to adjust this.
+	 */
+	public function exclude_from_sitemap( $exclude, $post ) {
+		$exclude = (bool) $exclude;
+		l( '$post<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<' );
+		l( $post );
+		$id = (int) get_option( 'wpcom_public_coming_soon_page_id', 0 );
+
+		if ( ! empty( $id ) && $post->ID === $id ) {
+			$exclude = (bool) apply_filters( 'wpcom_sitemaps_exclude_post', true, get_post( $id ) );
+
+		}
+
+		return $exclude;
+	}
+
+	public function exclude_from_public_queries( $query ) {
+		$id = (int) get_option( 'wpcom_public_coming_soon_page_id', 0 );
+		if ( ! empty( $id ) && ! is_admin() && $query->is_main_query() ) {
+			$query->set( 'post__not_in', array( $id ) );
+		}
+	}
+
+	public static function display_coming_soon_page() {
+		global $post;
+
+		$should_show_display_fallback_coming_soon_page = false;
+
+		if ( ! self::should_show_coming_soon_page() ) {
+			return;
+		}
+
+		$id = (int) get_option( 'wpcom_public_coming_soon_page_id', 0 );
+
+		if ( empty( $id ) ) {
+			$should_show_display_fallback_coming_soon_page = true;
+		}
+
+		$page = get_post( $id );
+
+		if ( ! $page || ! isset( $post ) ) {
+			$should_show_display_fallback_coming_soon_page = true;
+		}
+
+		// Disable a few floating UI things
+		add_filter( 'wpcom_disable_logged_out_follow', '__return_true', 1, 999 );
+		add_filter( 'wpl_is_enabled_sitewide', '__return_false', 1, 999 );
+		// add_filter( 'jetpack_disable_eu_cookie_law_widget', '__return_true', 1, 999 );
+
+		if ( $should_show_display_fallback_coming_soon_page ) {
+			self::render_fallback_coming_soon_page();
+		} else {
+			?><!doctype html>
+			<html <?php language_attributes(); ?>>
+				<head>
+					<meta charset="<?php bloginfo( 'charset' ); ?>" />
+					<meta name="viewport" content="width=device-width, initial-scale=1" />
+					<?php wp_head(); ?>
+				</head>
+				<body>
+					<?php
+						echo apply_filters( 'the_content', $page->post_content );
+					?>
+					<?php wp_footer(); ?>
+				</body>
+			</html>
+			<?php
+		}
+
+		die();
+	}
+
+	public static function render_fallback_coming_soon_page() {
+		remove_action( 'wp_enqueue_scripts', 'wpcom_actionbar_enqueue_scripts', 101 );
+		remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
+		remove_action( 'wp_print_styles', 'print_emoji_styles' );
+		remove_action( 'wp_head', 'header_js', 5 );
+		remove_action( 'wp_head', 'global_css', 5 );
+		remove_action( 'wp_footer', 'wpcom_subs_js' );
+		remove_action( 'wp_footer', 'stats_footer', 101 );
+		wp_enqueue_style( 'recoleta-font', '//s1.wp.com/i/fonts/recoleta/css/400.min.css' );
```

Comment:
No strong opinion on this, but you might also look at importing in CSS [from the typography package](https://github.com/Automattic/wp-calypso/tree/master/packages/typography).